feat(routing): add dynamic routing and basic page navigation

mahdisolgi1 opened the pull request:
- Enable navigation to empty pages: Home, Blog, and Posts/{post-slug}
- Implement dynamic routing based on environment variables (e.g., dev-tools route)

mahdisolgi1 committed:
feat(routing): add dynamic routing and basic page navigation

    - Enable navigation to empty pages: Home, Blog, and Posts/{post-slug}
    - Implement dynamic routing based on environment variables ( dev-tools route)

amirrezaDev1378 commented on frontend/src/pages/Root.tsx:
add a css class for styling instead of inline styling
@mahdisolgi1

amirrezaDev1378 commented:
I suggest to use module css:
https://vite.dev/guide/features#css-modules

amirrezaDev1378 commented:
Please apply types for all components and their props (if they have any)

amirrezaDev1378 commented:
check this out
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example

mahdisolgi1 committed:
refactor(css): Convert plain CSS to CSS Modules and implement a hybrid CSS approach

- Changed existing plain CSS to CSS Modules for better maintainability.
- Retained the index.css for global styles.
- Implemented a hybrid CSS approach to combine global styles with component-scoped styles.
- Added additional reset CSS to enhance consistency across components.

mahdisolgi1 committed:
refactor: Convert JavaScript components to TypeScript and add missing Home link in Navbar

- Changed all relevant components from JavaScript to TypeScript for improved type safety and maintainability.
- Added the missing Home link in the Navbar for better navigation.

mahdisolgi1 commented:
I made some changes in commits ca99293 and 389b476 that address the issues .

mahdisolgi1 commented:
If there’s anything more that can be implemented, please feel free to share

amirrezaDev1378 reviewed (approved):
LGTM!
